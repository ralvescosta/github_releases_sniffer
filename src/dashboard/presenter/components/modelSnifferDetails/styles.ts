import {StyleSheet} from 'react-native';
import {widthToDP} from '../../../../core/themes/size';

export const styles = StyleSheet.create({
  modalContainer: {
    flex: 1,
  },
  closeModal: {
    padding: widthToDP('3%'),
  },
  containerContent: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-around',
  },

  contentRepository: {
    width: '100%',
    alignItems: 'center',
  },
  repositoryOwnerAvatar: {
    width: 85,
    height: 85,
  },
  repositoryOwnerFakeAvatar: {
    width: 85,
    height: 85,
    backgroundColor: '#ccc',
  },
  repositoryText: {
    fontSize: widthToDP('8%'),
    fontWeight: 'bold',
    marginTop: widthToDP('2%'),
  },

  contentRelease: {
    width: '100%',
    alignItems: 'center',
  },
  releaseTitle: {
    fontSize: widthToDP('4.8%'),
  },
  releaseText: {
    fontSize: widthToDP('4.8%'),
    fontWeight: 'bold',
  },

  contentRemove: {
    width: '100%',
    height: 50,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    elevation: 5,
    backgroundColor: '#fff',
  },
});
