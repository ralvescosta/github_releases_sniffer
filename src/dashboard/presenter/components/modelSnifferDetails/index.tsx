import React from 'react';
import {Modal, View, TouchableOpacity, Text, Image} from 'react-native';

import {styles} from './styles';

import {IModalSnifferDetailsViewController} from '../../../interfaces/modalSnifferDetails/imodal.sniffer.details.view.controller';
import {SniffedGithubRepositoryEntity} from '../../../bussiness/entities/sniffed.github.repository.entity';

type Props = {
  repository: SniffedGithubRepositoryEntity;
  viewController: IModalSnifferDetailsViewController;
};

export const ModelSnifferDetails = ({repository, viewController}: Props) => {
  return (
    <Modal animationType="slide" visible={viewController.modalContext.toggleModal}>
      <View style={styles.modalContainer}>
        <TouchableOpacity style={styles.closeModal} onPress={() => viewController.closeModal()}>
          <Text>X</Text>
        </TouchableOpacity>
        <View style={styles.containerContent}>
          <View style={styles.contentRepository}>
            <Image style={styles.repositoryOwnerAvatar} source={{uri: repository.ownerAvatarUrl}} resizeMode="cover" />
            <Text style={styles.repositoryText}>{repository.fullName}</Text>
          </View>

          <View style={styles.contentRelease}>
            <Text style={styles.releaseTitle}> Latest release: </Text>
            <Text style={styles.releaseText}>{repository.lastRelease}</Text>
          </View>

          <TouchableOpacity onPress={() => viewController.removeRepository(repository.id)} style={styles.contentRemove}>
            <Text>Remove Repository on Sniffer</Text>
          </TouchableOpacity>
        </View>
      </View>
    </Modal>
  );
};
