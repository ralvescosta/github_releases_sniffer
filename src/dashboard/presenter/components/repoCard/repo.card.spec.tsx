import React from 'react';
import {render} from '@testing-library/react-native';

import {RepoCard} from './index';
import {ModelSnifferDetails} from '../modelSnifferDetails/index';
import {SniffedGithubRepositoryEntity} from '../../../bussiness/entities/sniffed.github.repository.entity';
import {RepoCardViewControllerSpy} from '../../__test__/repo.card.view.controller.spy';
import {ModalSnifferDetailsViewControllerSpy} from '../../__test__/modal.sniffer.details.view.controller.spy';

function makeSut() {
  const Wrapper = () => {
    const repository = new SniffedGithubRepositoryEntity(1, '', '', 10, 10, 10, 1, '', '', '');
    const cardControllerSpy = new RepoCardViewControllerSpy();
    const modalControllerSpy = new ModalSnifferDetailsViewControllerSpy();

    const WrapperModal = () => {
      return <ModelSnifferDetails repository={repository} viewController={modalControllerSpy} />;
    };
    return <RepoCard ModelSnifferDetails={WrapperModal} repository={repository} viewController={cardControllerSpy} />;
  };

  return render(<Wrapper />);
}

describe('Repo Card Components', () => {
  it('Should render correctly', () => {
    const sut = makeSut();

    expect(sut.toJSON()).toMatchSnapshot();
  });
});
