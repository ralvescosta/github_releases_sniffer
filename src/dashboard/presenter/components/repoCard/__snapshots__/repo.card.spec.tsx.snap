// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo Card Components Should render correctly 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
>
  <View
    accessible={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 6,
        "elevation": 5,
        "flexDirection": "row",
        "marginTop": 15,
        "opacity": 1,
        "paddingHorizontal": 26.5,
        "paddingVertical": 15,
        "width": "100%",
      }
    }
  >
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      visible={true}
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "padding": 22.5,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 34,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "space-around",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "width": "100%",
              }
            }
          >
            <Image
              resizeMode="cover"
              source={
                Object {
                  "uri": "",
                }
              }
              style={
                Object {
                  "height": 85,
                  "width": 85,
                }
              }
            />
            <Text
              style={
                Object {
                  "fontSize": 60,
                  "fontWeight": "bold",
                  "marginTop": 15,
                }
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 36,
                }
              }
            >
               Latest release: 
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 36,
                  "fontWeight": "bold",
                }
              }
            >
              
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "elevation": 5,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <Text>
              Remove Repository on Sniffer
            </Text>
          </View>
        </View>
      </View>
    </Modal>
    <View
      style={
        Object {
          "maxWidth": "60%",
          "width": "100%",
        }
      }
    >
      <View
        style={Object {}}
      >
        <Text
          numberOfLines={2}
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          
        </Text>
        <Text
          ellipsizeMode="tail"
          numberOfLines={3}
          style={
            Object {
              "color": "#777",
              "marginTop": 4,
            }
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 15,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "color": "#777",
              "flexDirection": "row",
              "marginRight": 7,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#777",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "fontSize": 25.5,
                "marginLeft": 3,
                "maxWidth": 40,
              }
            }
          >
            10
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "color": "#777",
              "flexDirection": "row",
              "marginRight": 7,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#777",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "fontSize": 25.5,
                "marginLeft": 3,
                "maxWidth": 40,
              }
            }
          >
            10
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "color": "#777",
              "flexDirection": "row",
              "marginRight": 7,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#777",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "fontSize": 25.5,
                "marginLeft": 3,
                "maxWidth": 40,
              }
            }
          >
            10
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "maxWidth": "40%",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Last Release: 
        </Text>
        <Text
          style={
            Object {
              "color": "#777",
            }
          }
        >
          
        </Text>
      </View>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#d4d4d4",
              "fontSize": 60,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RNGestureHandlerButton>
`;
